src/humanpy/hum_kin2.py

```diff
@@ -0,0 +1,298 @@
+                    comm_q3 = numpy.cos(q1)*numpy.cos(q3)*numpy.sin(q2)
+                    if side == 'L':                 
+                        val_q3 = (vect_norm_es_eh[1] + comm_q3)/numpy.sin(q1)                               
+                        q3_b = numpy.asscalar(numpy.arcsin(self.checkArg(val_q3))) 
+                    else:
+                        val_q3 = (-vect_norm_es_eh[1] + comm_q3)/numpy.sin(q1)
+                        q3_b = numpy.asscalar(numpy.arcsin(self.checkArg(val_q3))) 
+                    if q3_b < 0:                                                                            #[-pi,pi]
+                        q3 = -q3
+                else:
+                    q3 = None
+            else:
+                q1 = None
+                q3 = None
+
+            return [q1, q2, q3, q4]
+        else:
+            return None
+
+    def getLowerLimbAngles(self, tf, side):
+        if side == 'L':        
+            sys_hip = self.getSkeletonTransformation(tf, 'HipLeft', self.refsys)
+            sys_knee = self.getSkeletonTransformation(tf, 'KneeLeft', self.refsys)
+            sys_foot = self.getSkeletonTransformation(tf, 'AnkleLeft', self.refsys)
+        else:            
+            sys_hip = self.getSkeletonTransformation(tf, 'HipRight', self.refsys)
+            sys_knee = self.getSkeletonTransformation(tf, 'KneeRight', self.refsys)
+            sys_foot = self.getSkeletonTransformation(tf, 'AnkleRight', self.refsys)
+            
+        if sys_hip is None or sys_knee is None or sys_foot is None:
+            return None
+        
+        den_vect_kh = numpy.linalg.norm([sys_hip[0:3,3] - sys_knee[0:3,3]])
+        den_vect_fk = numpy.linalg.norm([sys_knee[0:3,3] - sys_foot[0:3,3]])
+        if numpy.abs(den_vect_kh) > 0.001 and numpy.abs(den_vect_fk) > 0.001:  
+            vect_kh = (sys_hip[0:3,3] - sys_knee[0:3,3])/den_vect_kh                   
+            vect_fk = (sys_knee[0:3,3] - sys_foot[0:3,3])/den_vect_fk   
+            q2 = - numpy.arccos(self.checkArg(numpy.asscalar(numpy.dot(vect_kh.T,vect_fk))))
+            
+            q1 = numpy.asscalar(numpy.arccos(vect_kh[1]))                                       #[0,pi]
+            if numpy.asscalar(numpy.arcsin(vect_kh[2])) < 0:                                    #[-pi,pi]
+                q1 = -q1 
+            
+            return [q1, q2]
+        else:
+            return None
+  
+    def checkLimits(self, joint, angle_vect, angle_elem):
+        if angle_vect is not None:
+            angle = angle_vect[angle_elem]
+            if angle is not None and joint is not None:
+                lower,upper = joint.GetLimits()
+                if angle < lower: angle = lower
+                if angle > upper: angle = upper
+                
+                self.currentDOFvalues[joint.GetDOFIndex()] = angle           
+                self.body.SetDOFValues(self.currentDOFvalues)
+    
+    def define_pose_from_eetransform(self, ee_pose):
```

Comment:
im not totally sure but are the 'define_pose_from_eetransform' and 'define_pose_from_pos_quat' general enough to go somewhere else like a util file?